File: java/demo/IceBox/hello/HelloI.java
// **********************************************************************
//
// Copyright (c) 2001
// ZeroC, Inc.
// Billerica, MA, USA
//
// All Rights Reserved.
//
// Ice is free software; you can redistribute it and/or modify it under
// the terms of the GNU General Public License version 2 as published by
// the Free Software Foundation.
//
// **********************************************************************

public class HelloI extends Hello
{
    public
    HelloI(Ice.Communicator communicator)
    {
        _communicator = communicator;
    }

    public void
    sayHello(Ice.Current current)
    {
        System.out.println("Hello World!");
    }

    public void
    shutdown(Ice.Current current)
    {
        System.out.println("Shutting down...");
        _communicator.shutdown();
    }

    private Ice.Communicator _communicator;
}

File: cpp/demo/IceBox/hello/HelloI.cpp
// **********************************************************************
//
// Copyright (c) 2001
// MutableRealms, Inc.
// Huntsville, AL, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <Ice/Ice.h>
#include <HelloI.h>

using namespace std;

HelloI::HelloI(const Ice::CommunicatorPtr& communicator) :
    _communicator(communicator)
{
}

void
HelloI::hello(const Ice::Current&)
{
    cout << "Hello World!" << endl;
}

void
HelloI::shutdown(const Ice::Current&)
{
    cout << "Shutting down..." << endl;
    _communicator->shutdown();
}
